### PR TITLE
feat: gw 7913 implement drag and drop to bookmark folder

### PR DESCRIPTION
**Pull request by mudana-grune**
https://youtrack.weseek.co.jp/issue/GW-7913
Implement drag and drop to bookmark item and folder

### CODE REVIEW

**Review by jam411 — CHANGES_REQUESTED** [review]
Please update Branch.
Please merge the latest "master" branch into "imprv/gw7848-imprv-bookmarks-content-on-user-page" to resolve the conflict.

Then merge the "imprv/gw7848-imprv-bookmarks-content-on-user-page" branch into "feat/gw-7913-implement-drag-and-drop-to-bookmark-folder" and see if there are any conflicts. Make sure there are no conflicts.

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx [hunk 1]
```diff
@@ -162,13 +167,87 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
...
+  const isDroppable = (item: BookmarkFolderItemProps, targetRoot: string, targetLevel: number): boolean => {
+    if (item.bookmarkFolder.parent === bookmarkFolder._id || item.bookmarkFolder._id === bookmarkFolder._id) {
+      return false;
+    }
+    if (item.root === targetRoot) {
+      if (item.level < targetLevel) {
+        return false;
+      }
+    }
+    return true;
```
Comment (jam411) [inline packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:219]:
```
if (item.root === targetRoot) {
  if (item.level < targetLevel) {
    return false;
  }
}
return true;
```

It is possible to write this way, if you like

`return item.root !== targetRoot || item.level >= targetLevel;`

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx [hunk 2]
```diff
@@ -236,7 +319,7 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
             </div>
             {hasChildren() && (
               <div className="grw-foldertree-count-wrapper">
-                <CountBadge count={ childCount } />
+                <CountBadge count={childCount} />
```
Comment (jam411) [inline packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:322]: ## Now
- Only folders one level down are counted
- Bookmark pages are not counted

image

## Request
- Is it possible to count and display only the number of all bookmarked pages under the distribution?
- Folders does not count.

Comment (jam411) [reply packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:322]: If you can impl that I will create new task.

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx [hunk 3]
```diff
@@ -162,13 +167,87 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
...
+      canDrag: monitor.canDrag(),
+    }),
+  });
+
+
+  const folderItemDropHandler = async(item: BookmarkFolderItemProps) => {
+    try {
+      await apiv3Put('/bookmark-folder', { bookmarkFolderId: item.bookmarkFolder._id, name: item.bookmarkFolder.name, parent: bookmarkFolder._id });
+      await mutateChildBookmarkData();
+      toastSuccess(t('toaster.update_successed', { target: t('bookmark_folder.bookmark_folder') }));
```
Comment (jam411) [inline packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:190]: ## Check Request
i18n is not working if use `bookmark_folder.bookmark_folder`, please check grep: `bookmark_folder`.
Now `bookmark_folder` json is in the translation.json. But some toastSuccess use `ns: 'commons'`.
ex. `toastSuccess(t('toaster.update_successed', { target: t('bookmark_folder.bookmark_folder'), ns: 'commons' }));`

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx (hunk 2 above)
Comment (mudana-grune) [reply packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:322]: Yes it's possible to count all bookmarked pages

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx (hunk 1 above)
Comment (mudana-grune) [reply packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx]: I just implemented this line change

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx (hunk 2 above)
Comment (jam411) [reply packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:317]: I will create new task.

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx (hunk 3 above)
Comment (jam411) [reply packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx]: @mudana-grune 
It is not working. Please check be careful.
image

File: packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx [hunk 4]
```diff
@@ -198,25 +198,20 @@ const BookmarkFolderItem: FC<BookmarkFolderItemProps> = (props: BookmarkFolderIt
     try {
       await apiv3Post('/bookmark-folder/add-boookmark-to-folder', { pageId: item._id, folderId: bookmarkFolder._id });
       mutateParentBookmarkFolder();
-      toastSuccess('Bookmark added to bookmark folder successfully');
+      toastSuccess(t('toaster.add_successed', { target: t('bookmark_folder.bookmark'), ns: 'commons' }));
```
Comment (jam411) [inline packages/app/src/components/Bookmarks/BookmarkFolderItem.tsx:201]: You should use `toaster.add_succeeded`.

**Review by jam411 — CHANGES_REQUESTED** [review]

**Pull request merged**